### PR TITLE
polarization angle marginalization in single_template

### PR DESCRIPTION
**Pull request by shobhitrphy**

### CODE REVIEW

File: pycbc/inference/models/single_template_p.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        vloglr += abs(shloglr) + hhloglr 
+        #putting all vloglr values in exp(vloglr)
+        temp=numpy.exp(vloglr)
+        #summing over the values in the array exp(vloglr)
+        loglm=(numpy.sum(temp))
+        
+        #To handle the case where loglm=0; assigned a very small value to loglm that is not equal to 0
+        #taking the log of loglm to obtain new loglikelihood ratio marginalized over pol
+        
+        if loglm!=0:
```
Comment (ahnitz) [inline pycbc/inference/models/single_template_p.py:120]: I suspect you should not need this switch here if you implement the above. The reason you were probably getting identically zero is due to numerical error.

File: pycbc/inference/models/single_template_p.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            sh = self.sh[ifo].at_time(p['tc'] + dt) * htf
+            shloglr += sh
+            hhloglr += self.hh[ifo] * abs(htf) ** 2.0
+
+        vloglr = numpy.log(scipy.special.i0e(abs(shloglr)))
+        
+        #An array of vloglr for each polarization value
+        vloglr += abs(shloglr) + hhloglr 
+        #putting all vloglr values in exp(vloglr)
+        temp=numpy.exp(vloglr)
```
Comment (ahnitz) [inline pycbc/inference/models/single_template_p.py:113]: Line 113 and 115 should be combined into a single step with https://docs.scipy.org/doc/scipy-0.19.0/reference/generated/scipy.misc.logsumexp.html. It has much better accuracy than taking an exponential directly then summing.

File: pycbc/inference/models/single_template_p.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        p = self.current_params.copy()
+        p.update(self.static_params)
+
+        if self.time is None:
+            self.time = p['tc']
+
+        shloglr = hhloglr = 0
+        for ifo in self.sh:
+            fp, fc = self.det[ifo].antenna_pattern(p['ra'], p['dec'],
+                                                   p['polarization'],
```
Comment (ahnitz) [inline pycbc/inference/models/single_template_p.py:94]: It appears you are still getting polarization from the param dict. For marginalization, polarization really shouldn't be a parameter anymore, instead you should make the vector in the init of this function (with some option for how finely it is sampled). Just make a vector that spans  from 0 to 2pi with a configureable number of grid points.

**Comment by ahnitz** [comment]
@shobhitrphy Also, it may be good to implement this as an option to the regular single template class, i.e. you set say marginalize_polarization=NUM_SAMPLES and then it does so as you've implemented, but otherwise it does what the code used to do and you can sample over polarization (or use a fixed value) as you like.

File: pycbc/inference/models/single_template_p.py (hunk 2 above)
Comment (shobhitrphy) [reply pycbc/inference/models/single_template_p.py:113]: Alright. Will implement that.

**Comment by ahnitz** [comment]
@shobhitrphy I think one of the Travis failures was just intermittent, if this passes the automated tests, I think this might be close to merging.

**Review by ahnitz — APPROVED** [review]

**Pull request merged**